maven-core/src/main/java/org/apache/maven/model/plugin/DefaultLifecycleBindingsInjector.java

```diff
@@ -87,18 +102,21 @@ else if ( !defaultPlugins.isEmpty() )
     {
 
         private static final String PLUGIN_MANAGEMENT = "plugin-management";
+        private static final String NO_VERSION_PLUGINS = "no-version-plugins";
 
-        public void merge( Model target, Model source )
+        public Collection<Plugin> merge( Model target, Model source )
```

Comment:
This looks weird: it says: merge 2 models, the result is a Collection of Plugins.

Reply:
true, I forgot to add at least a javadoc to explain the meaning of the return value, which was my minimal idea
reading your comment, I just found another idea: let's pass ModelProblemCollector
I like team work :)